test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpExporterTest.cs

```diff
@@ -285,10 +322,49 @@ public void UseOpenTelemetryProtocolActivityExporterWithCustomActivityProcessor(
             RepeatedField<OtlpCommon.KeyValue> otlpAttributes)
         {
             Assert.Equal(expectedAttributes.Count(), otlpAttributes.Count);
+            string lastKey = null;
+            int index = 0;
             for (int i = 0; i < otlpAttributes.Count; i++)
             {
                 Assert.Equal(expectedAttributes[i].Key, otlpAttributes[i].Key);
-                AssertOtlpAttributeValue(expectedAttributes[i].Value, otlpAttributes[i]);
+
+                if (string.IsNullOrEmpty(lastKey))
+                {
+                    lastKey = otlpAttributes[i].Key;
+                }
+
+                if (lastKey == otlpAttributes[i].Key)
+                {
+                    index++;
+                }
+                else
+                {
+                    index = 0;
+                }
+
+                if (expectedAttributes[i].Value.GetType().IsArray)
```

Comment:
Consider extracting `expectedAttributes[i].Value` to a variable to save some code.